// vite-plugin-define/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function definePlugin(options = {}) {
    return {
        name: "vite-plugin-define",
        transformIndexHtml(html, { server }) {
            const { ENVNAME = "BUILD_ENV", debug = true } = options;
            const isDev = server &&
                server.config &&
                server.config.command &&
                server.config.command === "serve";
            const CURBUILD_ENV = process.env[ENVNAME] || "local";
            const devScript = `window['${ENVNAME}']='${CURBUILD_ENV}';`;
            const buildScript = `var host=window.location.host||'';var domain_prefix=host.split('.')[0]||'';if(domain_prefix.indexOf('dev-')>-1){window['${ENVNAME}']='dev';} else if(domain_prefix.indexOf('sit-')>-1){window['${ENVNAME}']='sit';}else if(domain_prefix.indexOf('test-')>-1){window['${ENVNAME}']='test';}else if(domain_prefix.indexOf('pre-')>-1){window['${ENVNAME}']='pre';}else if(domain_prefix.indexOf('perf-')>-1){window['${ENVNAME}']='perf';}else{window['${ENVNAME}']='prod';}`;
            const debugScript = `var script=document.createElement("script");script.type="text/javascript";script.src="https://test-static.bankneo.co.id/oao/debug/eruda-1.5.4/eruda.min.js";script.onload=function(){window.eruda.init();};var head=document.getElementsByTagName('head')[0];head.appendChild(script);`;
            const tags = [
                {
                    tag: "script",
                    children: `${isDev ? devScript : buildScript}if(window['${ENVNAME}']!=='prod'){console.log('${ENVNAME}: '+window['${ENVNAME}']);${debug ? debugScript : ""}}`,
                    injectTo: "head-prepend",
                    attrs: {
                        defer: true,
                        importance: "low",
                    },
                },
            ];
            return { html, tags };
        },
    };
}
exports.default = definePlugin;
